locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/JcloudsSshMachineLocation.java

```diff
@@ -162,8 +162,12 @@ public void init() {
     @Override
     public void rebind() {
         super.rebind();
-        ComputeServiceContext context = jcloudsParent.getComputeService().getContext();
-        runScriptFactory = context.utils().injector().getInstance(RunScriptOnNode.Factory.class);
+        
+        if (jcloudsParent != null) {
+            // can be null on rebind, if location has been "orphaned" somehow
+            ComputeServiceContext context = jcloudsParent.getComputeService().getContext();
+            runScriptFactory = context.utils().injector().getInstance(RunScriptOnNode.Factory.class);
+        }
```

Comment:
@neykov Yes, it would - if the location were ever used. I'll add a log message in the else block.

The current behaviour is that Brooklyn fails to rebind because of the NPE, unless you configure it to ignore errors. It happened for some persisted state that had an "orphaned" location in its persisted state.

The ideal would be that locations are entities, and that the location can thus be marked as "on-fire". But that's much longer term.